addons/stock_account/models/stock.py

```diff
@@ -87,27 +87,54 @@ def _get_price_unit(self):
             move = self.search([('product_id', '=', self.product_id.id),
                          ('state', '=', 'done'), 
                          ('location_id.usage', '=', 'internal'), 
-                         ('location_dest_id.usage', '!=', 'internal')], order='date desc', limit=1)
+                         ('location_dest_id.usage', '!=', 'internal'),
+                         ('company_id', '=', self.company_id.id)], order='date desc', limit=1)
```

Comment:
should probably expose this domain, as it's the same domain as _get_candidates_out_move but with a different order and limit